### DISCUSSION TITLE
Thanks.

### DISCUSSION THREAD

**Discussion by BinaryGenius** [zb3/gnome-gamma-tool#2] [General]
Just wanted to say I've been having this ongoing issue for months relating to brightness control. I use xrandr to change the brightness on Xorg so by itself that works fine. However with night light on gnome it doesn't work every time the temperature changes the brightness shoots back up and scorches my eye balls. I used to use a script with gnome and redshift but the issue with that was it randomly turned off when any application wanted to mess with my gamma. Also also there was no smooth transition with the temp and it didn't change when the sun set I just set keybinds to change the command to also change temp after the sun set. It was this whoooole big mess. But your script solved that completely it even solved another issue of remembering my settings past reboot!!! My eyes and my sleep schedule thank you. If you want my keybind script that I use on gnome for changing the brightness with your script I'd be happy to send it over.

**Comment by zb3** [comment]
@BinaryGenius, I'm happy to hear that the tool was helpful :)
It's also good to know it could be found, as I don't have any marketing capabilities.

I still think that the option to change gamma/contrast/brightness should be built-in into GNOME (yet scriptable), the lack of this really surprised me when I first installed Fedora, hence I created this.

Feel free to post your script here, it could possibly help others :)

**Reply by BinaryGenius** [reply]
Yeah I feel the same way we should at least have an extension for brightness control. It amazes me that such an integral part of computing is left out of one of the biggest desktops for linux. I changed brightness.sh >> brightness.txt and your script to a .txt as well so github would let me upload it here. I add a keyboard command to run the script in gnome settings it's something like "sh /home/(username)/brightness.sh Up" and then with the "Down" argument to lower brightness. Mapped to Ctrl F10 and F9 respectively.
[brightness.txt](https://github.com/zb3/gnome-gamma-tool/files/8690065/brightness.txt)
[brightness-level.txt](https://github.com/zb3/gnome-gamma-tool/files/8690066/brightness-level.txt)
[gnome-gamma-tool.txt](https://github.com/zb3/gnome-gamma-tool/files/8690067/gnome-gamma-tool.txt)
This definitely needs work because on Fedora 35 if you use the keys too much it brings up a "Oops!" notification in problem reporting it also may causes weird behavior that I don't know how to fix.

**Reply by zb3** [reply]
Thanks for posting your scripts. 

One possibility is that when you're pressing those keys too fast, there might be some race condition that causes ggt to crash because of an unsuccessful call to `colord`.

If you could attach ggt output, I could try to see if that's the case. If you don't see the output, maybe catch it via `tee` in your script:
```
python $gnomegammatoolpath -b $brightness -y |& tee -a /tmp/ggt-log
```
and then see the `/tmp/ggt-log` file

(also note Fedora 36 was recently released)

I'm not sure how it works in GNOME 42, but at the time of creating this tool, I saw that it wasn't possible to change gamma/brightness/contrast using an extension without using color profiles (the gamma table was computed by mutter directly using the color profile and the night shift option value).
That's bad because you can't do a GUI slider with smooth realtime adjustment since every change needs a new color profile and this is slow/inefficient.

There was a (privileged I suppose) mutter DBus call to set the gamma table directly and maybe that could work for the slider, yet obviously that's not permanent, so color profiles are unavoidable here.

**Reply by BinaryGenius** [reply]
There probably is a race condition I only get the bug when I press the keys too fast. But idk how to limit a users presses like somehow put a timer that only allows them to press after 1 second? I don't want the brightness to change too slowly. I also still want the brightness to be capped at 0.1 so it doesn't black out the screen and 1 so it doesn't over saturate the screen. As for the extension I don't think it needs to be smooth it can be sort of segmented where the slider locks at certain values. If there could be any sort of extension made no matter how bad I think that could be huge for a lot of users.
### As for the ggt thing I have no idea what that is please explain I'm fairly new to this.

**Reply by BinaryGenius** [reply]
Have you given up on the code?

**Reply by zb3** [reply]
Ah I didn't see your edit so I thought it was your turn... I only read mails and edits aren't included there and as we can see, maybe they should..

Now, in order to be able to try to fix the error message, I need to know what it is (now we can only guess), so I need you to provide the log and then file an issue.
**By "ggt" I meant "gnome-gamma-tool"** :)

So in your script, append `|& tee -a /tmp/ggt-log` to the calls to the tool. So if you had:
```
python $gnomegammatoolpath -b $brightness -y
```
replace this with:
```
python $gnomegammatoolpath -b $brightness -y |& tee -a /tmp/ggt-log
```
and each time you have this `python $gnomegammatoolpath`, append `|& tee -a /tmp/ggt-log` to that line.

Then *after* you see the error (and only in that case), please file an issue with the contents of the `/tmp/ggt-log` file if they contain the error message (otherwise we need some other method of catching this)

I hope this helps, but let me know if something is still unclear

**Reply by BinaryGenius** [reply]
I'll see what I can do thanks for the insight I'll be back asap.

**Comment by SlappyHours** [comment]
I just wanna say thanks as well, @zb3 . The tool really helped correcting the colors of my notebook display. I can't believe this feature is still missing in GNOME.
